src/uct/tcp/tcp_cm.c

```diff
@@ -199,11 +201,11 @@ ucs_status_t uct_tcp_cm_send_event(uct_tcp_ep_t *ep, uct_tcp_cm_conn_event_t eve
         uct_tcp_cm_trace_conn_pkt(ep, UCS_LOG_LEVEL_TRACE,
                                   "%s sent to", event);
     } else {
+        ucs_assert(status != UCS_ERR_NO_PROGRESS);
         status = uct_tcp_ep_handle_io_err(ep, "send", status);
         uct_tcp_cm_trace_conn_pkt(ep,
-                                  (((status == UCS_ERR_NO_PROGRESS) ||
-                                    (status == UCS_ERR_CANCELED)) ?
-                                   UCS_LOG_LEVEL_DEBUG : UCS_LOG_LEVEL_ERROR),
+                                  (log_error && (status != UCS_ERR_CANCELED)) ?
+                                  UCS_LOG_LEVEL_DEBUG : UCS_LOG_LEVEL_ERROR,
```

Comment:
maybe add local var log_level?

Reply:
but we still need to check for `status` that is set during `ucs_socket_send()`